modules/core/core-elasticsearch-server/src/main/java/com/enonic/xp/elasticsearch/server/config/ElasticsearchServerConfigResolver.java

```diff
@@ -0,0 +1,279 @@
+package com.enonic.xp.elasticsearch.server.config;
+
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+
+import com.enonic.xp.elasticsearch.server.impl.ElasticsearchServerSettings;
+import com.enonic.xp.home.HomeDir;
+
+import static com.google.common.base.Strings.nullToEmpty;
+
+public final class ElasticsearchServerConfigResolver
+{
+
+    private final ElasticsearchServerConfig configuration;
+
+    public ElasticsearchServerConfigResolver( final ElasticsearchServerConfig configuration )
+    {
+        this.configuration = configuration;
+    }
+
+    public ElasticsearchServerSettings resolve()
+    {
+        final ElasticsearchConfigValidator validator = new ElasticsearchConfigValidator();
+        validator.validate();
+
+        final ElasticsearchServerSettings.Builder builder = ElasticsearchServerSettings.builder().
+            pathConf( resolvePathConfDir() ).
+            pathData( resolvePathDataDir() ).
+            pathLogs( resolvePathLogsDir() ).
+            pathRepo( resolvePathRepoDir() ).
+            pathWork( resolvePathWorkDir() ).
+            gatewayExpectedNodes( configuration.gateway_expectedNodes() ).
+            gatewayRecoverAfterNodes( configuration.gateway_recoverAfterNodes() ).
+            gatewayRecoverAfterTime( configuration.gateway_recoverAfterTime() ).
+            clusterRoutingAllocationDiskThresholdEnabled( configuration.cluster_routing_allocation_disk_thresholdEnabled() ).
+            clusterName( configuration.cluster_name() ).
+            indexMaxResultWindow( configuration.index_maxResultWindow() ).
+            httpPort( configuration.http_port() ).
+            transportPort( configuration.transport_port() );
+
+        return builder.build();
+    }
+
+    public String resolveElasticServerDir()
```

Comment:
A lot of similar code above. Check for empty and resolve relative path could be in private function